Css frameworks submission

JoranEngelund opened the pull request.

JoranEngelund committed:
setup bootstrap, sass, live-server

JoranEngelund committed:
setup src/scss/styles.scss & imported bootstrap.scss and functions.scss / compiled it to dist/css/styles.css

JoranEngelund committed:
added profile-image "placeholder" in assets folder

JoranEngelund committed:
added logo.svg to assets folder

JoranEngelund committed:
css with compiled scss styling

JoranEngelund committed:
added login page, styled with bootstrap, scss

JoranEngelund committed:
customized bs-btn padding, border and border radius

JoranEngelund committed:
customized bs-card, border-radius, width, bg-color

JoranEngelund committed:
created custom mixin for images

JoranEngelund committed:
customized bs-colors, box-shadow, spacers, link-color

JoranEngelund committed:
customized bs-body-font-family, styled img-logo,  imported font, imported custom scss files

JoranEngelund committed:
Update styles.css.map

JoranEngelund committed:
removed custom padding

JoranEngelund committed:
added customized bootstrap padding on button

JoranEngelund committed:
compiled scss into css

JoranEngelund committed:
customized .nav-link class

JoranEngelund committed:
customized nav-link font-size

JoranEngelund committed:
customized bs-border-radius

JoranEngelund committed:
customized -bs-border-radius

JoranEngelund committed:
added and styled home.html file

JoranEngelund committed:
built home.html page

JoranEngelund committed:
change margin-top on card-title and added custom class on login-btn

JoranEngelund committed:
compiled scss to css

JoranEngelund committed:
customized card-space-y and customized card class

JoranEngelund committed:
customized navbar-focus on icon

JoranEngelund committed:
added extra spacing on .m / .p classes on

JoranEngelund committed:
added customization on bootstrap classes

customized border-width in :root
updated some structures regarding DRY
customized form-control
added custom style for btn-login
customized .card
customized card-img-top class

JoranEngelund committed:
added class sticky-top for header

JoranEngelund committed:
added text-decoration-underline for anchor tag

JoranEngelund committed:
Updated folder and file structures for custom scss files

JoranEngelund committed:
changed structure of imports, customized bs classes

changed import structures
customized h2 font-size within cards
customized .card-img-bottom styling

JoranEngelund committed:
further customizing of spacing variables

JoranEngelund committed:
updated form action href

JoranEngelund committed:
updated nav link href to profile, changed padding on submit btn

JoranEngelund committed:
added profile page, done styled

JoranEngelund committed:
compiled scss to css

JoranEngelund committed:
updated meta description content

JoranEngelund committed:
added favicons for each page

JoranEngelund committed:
changed input ="text" to <textarea>

JoranEngelund committed:
changed padding on submit btn from 7-8

JoranEngelund committed:
removed margin-start from div container

removed margin-start from div containerr for profile image + name as it caused a horizontal scroll-bar

Fermain commented on src/scss/styles.scss:
```diff
@@ -0,0 +1,106 @@
+
+// Customized form from "../../node_modules/bootstrap/scss/forms/form-control.scss"
+.form-control {
+  width: 17rem;
```
```suggestion
  width: 100%;
  max-width: 17rem;
```

This will give better responsive behaviour. Setting a maximum bound will allow for the element to shrink when needed.

JoranEngelund replied:
I'll fix this at once!

JoranEngelund replied:
This has now been updated

Fermain commented on src/scss/styles.scss:
```diff
@@ -0,0 +1,106 @@
+  &.search-bar {
+    font-size: 1.5rem;
+    height: 3rem;
+    width: 13rem;
```
Same here, max-width is more tolerant

JoranEngelund replied:
thanks! fixed it now

Fermain commented on src/scss/styles.scss:
```diff
@@ -0,0 +1,106 @@
+  }
+
+  &.post-input {
+    width: 19rem;
```
And here :)

JoranEngelund replied:
thanks! fixed it now

Fermain reviewed (changes requested):
There are some very minor best practice improvements to be made, but aside from these this submission shows a deep understanding of SASS and Bootstrap customisation. The site meets the brief well and with these minor changes applied will be ready for delivery.

If you haven't already, it is worth posting on the forum for peer review and offering a review to another student. This is something you will be doing continuously throughout this year so it's worth getting into the swing of things now.

Well done.

JoranEngelund committed:
updated width per review request

updated maximum bound on form-control

JoranEngelund committed:
compiled css

JoranEngelund committed:
updated per review request

updated maximum bound on .post-input and .search-bar

JoranEngelund committed:
changed h1 to h2

there were two h1's on one page. needed to change one of them

JoranEngelund committed:
updated form child elements

removed label as h1 was not allowed to be child of labels
removed type attribute from textarea element

JoranEngelund committed:
Removed h2 from label

Removed h2 from label since not allowed to be child for labels. added text left on label class to maintain left aligned when responsive

JoranEngelund committed:
removed type attribute from textarea

JoranEngelund committed:
changed margin and padding start-end to x

changed margin and padding start-end to x, to utilize x-axis classes for better readability

JoranEngelund committed:
changed margin and padding start-end, top-bottom to m/p-y and m/p-x

JoranEngelund committed:
updated .p start-end to px

JoranEngelund committed:
updated nav-link padding and margins classes from start/end top/bottom to x, y

JoranEngelund committed:
updated button padding from start/end to px-#

JoranEngelund committed:
installed bootstrap icons
